src/ansys/grantami/bomanalytics/_connection.py

```diff
@@ -210,14 +210,47 @@ def __init__(self, servicelayer_url: str, **kwargs: Any) -> None:
             "substances_table_name": None,
             "coatings_table_name": None,
         }
+        self._max_spec_depth: Optional[int] = None
 
     def __repr__(self) -> str:
-        base_repr = f'<BomServicesClient: url="{self._sl_url}", dbkey="{self._db_key}"'
-        custom_tables = ", ".join([f'{k}="{v}"' for k, v in self._table_names.items() if v])
-        if custom_tables:
-            return base_repr + f", {custom_tables}>"
-        else:
-            return base_repr + ">"
+        max_link_value: Union[str, int] = (
+            "unlimited" if self.maximum_spec_link_depth is None else self.maximum_spec_link_depth
+        )
+        repr_entries: List[Tuple[str, Union[str, int, None]]] = [
+            ("url", self._sl_url),
+            ("maximum_spec_link_depth", max_link_value),
+            ("dbkey", self._db_key),
+        ]
+        for k, v in self._table_names.items():
+            if v:
+                repr_entries.append((k, v))
+        rendered_entries = []
+        for name, value in repr_entries:
+            if isinstance(value, str):
+                value = f'"{value}"'
+            rendered_entries.append(f"{name}={value}")
+        return f'<BomServicesClient: {", ".join(rendered_entries)}>'
+
+    @property
+    def maximum_spec_link_depth(self) -> Optional[int]:
+        """Limits the maximum depth of specification to specification links that will be followed when determining
```

Comment:
The wording here is really hard to get right. To be both precise and concise. How about:

"Limits the maximum number of specification-to-specification links that will be followed when processing a query. If specified, specification-to-specification links will be truncated at the specified depth, and only coatings and substances identified up to and including that point will be included in the analysis.

Defaults to ``None``, which applies no limit to the number of specification-to-specification links. This may lead to performance issues if there are large numbers of specification-to-specification links present in the database.

Note: this limit applies to each branch of the BoM individually. This is not a global limit on the number of specification-to-specification links that will be traversed across the entire BoM, instead it is a limit on the maximum depth of specifications below any individual specification node."